com.microsoft.java.debug.plugin/src/main/java/com/microsoft/java/debug/LambdaExpressionLocator.java

```diff
@@ -0,0 +1,97 @@
+/*******************************************************************************
+* Copyright (c) 2022 Microsoft Corporation and others.
+* All rights reserved. This program and the accompanying materials
+* are made available under the terms of the Eclipse Public License v1.0
+* which accompanies this distribution, and is available at
+* http://www.eclipse.org/legal/epl-v10.html
+*
+* Contributors:
+*     Gayan Perera (gayanper@gmail.com) - initial API and implementation
+*******************************************************************************/
+package com.microsoft.java.debug;
+
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.ASTVisitor;
+import org.eclipse.jdt.core.dom.AbstractTypeDeclaration;
+import org.eclipse.jdt.core.dom.CompilationUnit;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.LambdaExpression;
+
+public class LambdaExpressionLocator extends ASTVisitor {
+    private CompilationUnit compilationUnit;
+    private int line;
+    private int column;
+    private boolean found;
+
+    private IMethodBinding lambdaMethodBinding;
+    private LambdaExpression lambdaExpression;
+
+    public LambdaExpressionLocator(CompilationUnit compilationUnit, int line, int column) {
+        this.compilationUnit = compilationUnit;
+        this.line = line;
+        this.column = column;
+    }
+
+    @Override
+    public boolean visit(LambdaExpression node) {
+        if (column > -1) {
+            int startPosition = node.getStartPosition();
+
+            int columnNumber = this.compilationUnit.getColumnNumber(startPosition);
+            int lineNumber = this.compilationUnit.getLineNumber(startPosition);
+
+            if (column == columnNumber && lineNumber == line) {
```

Comment:
Currently the lambda inline breakpoint must be set at the beginning of the lambda expression, which is not friendly for lambda that uses parentheses to wrap args.

`list.stream().forEach(▮(name) -> System.out.println(name))` works, but 
`list.stream().forEach((▮name) -> System.out.println(name))` does not.

I'm wondering whether we should relax the lambda breakpoint condition, e.g.  
the column breakpoint is valid as long as it stops within the lambda parameter list.

@gayanper WDYT?

Reply:
Yes i think that make more sense, I can update the criteria to support it.

Reply:
Try now @testforstephen